packages/react-spinbutton/Spec.md

```diff
@@ -2,62 +2,381 @@
+- [Allowing Decimal Values](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#allowing_decimal_values)
+- [Offering Suggested Values](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#offering_suggested_values)
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+### Uncontrolled Example
+
+```tsx
+<SpinButton label="Uncontrolled Example" defaultValue="1" />
+```
+
+### Controlled Example
+
+```tsx
+type SpinButtonChangeData = {
+  value: number; // The "raw value", e.g., 2
+  formattedValue: string; // The "formatted value", e.g., "$2.00"
+  prevValue: number;
+  prevFormattedValue: string;
+};
+
+const [value, setValue] = useState<number>(2);
+const onControlledExampleChange = (_event, data: SpinButtonChangeData) => {
+  setValue(data.value);
+};
+
+<SpinButton label="Controlled Example" value={value} onChange={onControlledExampleChange} />;
+```
+
+### Custom Display Format Example
+
+```tsx
+type SpinButtonChangeData = {
+  value: number;
+  formattedValue: string;
+  prevValue: number;
+  prevFormattedValue: string;
+};
+
+type SpinButtonFormatter = (value: number) => string;
+type SpinButtonParser = (formattedValue: string) => number;
+
+const [value, setValue] = useState<number>(3);
+const onControlledExampleChange = (_event, data: SpinButtonChangeData) => {
+  setValue(data.value);
+};
+
+// Takes a number in and returns a formatted string
+// Ex: 12 becomes "12 pt"
+const fontFormatter: SpinButtonFormatter = value => {
+  return `${value} pt`;
+};
+
+// Takes a formatted string in and returns a number
+// Ex: "12 pt" becomes 12
+const fontParser: SpinButtonParser = formattedValue => {
+  return parseFloat(formattedValue);
+};
+
+<SpinButton
+  label="Controlled Formatted Example"
+  value={value}
+  formatter={fontFormatter}
+  parser={fontParser}
+  onChange={onControlledExampleChange}
+/>
+
+<SpinButton
+  label="Uncontrolled Formatted Example"
+  defaultValue={4}
+  formatter={fontFormatter}
+  parser={fontParser}
+/>
+```
+
+### Basic Example Implementation
+
+A very basic example to demonstrate how formatting will work in practice.
+
+[Link to example on Codesandbox](https://codesandbox.io/s/spinbutton-example-formatted-values-66pou?file=/src/SpinButton/SpinButton.js)
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+[The native HTML number input can be associated with a list](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#offering_suggested_values) to provide suggested values in the `SpinButton`. This fits nicely when a scenario where a `SpinButton` is used as a font size selector however it seems to have some overlap with a `ComboBox`. As the existing v8 `SpinButton` lacks this feature it will be left off the initial implementation.
```

Comment:
Is the autocomplete variant actually specced by design ? If not we should just leave it out of our spec completely